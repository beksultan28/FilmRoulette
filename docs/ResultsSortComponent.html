<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: ResultsSortComponent</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: ResultsSortComponent</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>ResultsSortComponent<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="ResultsSortComponent"><span class="type-signature"></span>new ResultsSortComponent<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Component for sorting search results
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_results-sort_index.jsx.html">components/results-sort/index.jsx</a>, <a href="components_results-sort_index.jsx.html#line6">line 6</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    
        <h3 class="subsection-title">Extends</h3>

        


    <ul>
        <li>Component</li>
    </ul>


    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id=".handleSortByChange"><span class="type-signature">(static) </span>handleSortByChange<span class="signature">(event)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    onShort change event handlerStops event bubble and calls onSortproperty function with event data
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>event</code></td>
            

            <td class="type">
            
                
<span class="param-type">Object</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_results-sort_index.jsx.html">components/results-sort/index.jsx</a>, <a href="components_results-sort_index.jsx.html#line28">line 28</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="DetailComponent.html">DetailComponent</a></li><li><a href="ResultsListComponent.html">ResultsListComponent</a></li><li><a href="ResultsSortComponent.html">ResultsSortComponent</a></li><li><a href="SearchFormComponent.html">SearchFormComponent</a></li></ul><h3>Global</h3><ul><li><a href="global.html#apiKey">apiKey</a></li><li><a href="global.html#detailItem">detailItem</a></li><li><a href="global.html#expectedResult">expectedResult</a></li><li><a href="global.html#expectedResults">expectedResults</a></li><li><a href="global.html#FooterComponent">FooterComponent</a></li><li><a href="global.html#getFilmApi">getFilmApi</a></li><li><a href="global.html#getFilmById">getFilmById</a></li><li><a href="global.html#getRecommendations">getRecommendations</a></li><li><a href="global.html#getRecommendationsApi">getRecommendationsApi</a></li><li><a href="global.html#recieveFilmById">recieveFilmById</a></li><li><a href="global.html#recieveFilms">recieveFilms</a></li><li><a href="global.html#ResultsItemComponent">ResultsItemComponent</a></li><li><a href="global.html#searchApi">searchApi</a></li><li><a href="global.html#searchBy">searchBy</a></li><li><a href="global.html#searchFilms">searchFilms</a></li><li><a href="global.html#searchResults">searchResults</a></li><li><a href="global.html#sortBy">sortBy</a></li><li><a href="global.html#sortFilms">sortFilms</a></li><li><a href="global.html#updateQuery">updateQuery</a></li><li><a href="global.html#updateSearchBy">updateSearchBy</a></li><li><a href="global.html#updateSortBy">updateSortBy</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Sat Oct 28 2017 22:06:51 GMT+0600 (Central Asia Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>