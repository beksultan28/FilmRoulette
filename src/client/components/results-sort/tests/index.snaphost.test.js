import React from 'react';
import renderer from 'react-test-renderer';

import ResultsSortComponent from '../index.jsx';

const onSort = jest.fn();
const props = {
  title: 'Test title',
  sortBy: 'release_date',
  onSort,
};
const resultSort = renderer.create(
  <ResultsSortComponent
    {...props}
  />,
);

test('ResultsSortComponent renders correctly', () => {
  expect(resultSort.toJSON()).toMatchSnapshot();
});
