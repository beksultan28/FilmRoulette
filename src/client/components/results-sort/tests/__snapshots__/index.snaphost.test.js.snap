// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsSortComponent renders correctly 1`] = `
<div
  className="results-sort"
>
  <div
    className="container wrapper"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-6 results-sort-title"
      >
        Test title
      </div>
      <div
        className="col-xs-12 col-sm-6 sort-by-block"
      >
        Sort by:
        <a
          className="active"
          data-value="release_date"
          href="#!"
          onClick={[Function]}
        >
          release date
        </a>
        <a
          className=""
          data-value="vote_average"
          href="#!"
          onClick={[Function]}
        >
          rating
        </a>
      </div>
    </div>
  </div>
</div>
`;
